pkg/tests/util/defs.bzl

```diff
@@ -58,6 +58,44 @@ Paths containing directories will also have the intermediate directories created
     },
 )
 
+def _fake_artifact_impl(ctx):
+    out_file = ctx.actions.declare_file(ctx.attr.name)
+    content = ['echo ' + rf.path for rf in ctx.files.runfiles]
+    ctx.actions.write(
+        output = out_file,
+        content = '\r\n'.join(content),
+        is_executable = ctx.attr.executable,
+    )
+    return DefaultInfo(
+        files = depset([out_file]+ctx.files.runfiles),
+        executable = out_file if ctx.attr.executable else None,
+    )
+
+fake_artifact = rule(
+    doc = """Rule to create a fake artifact that depends on its srcs.
+
+This rule creates a file that appears to depend on its srcs and passes along
+some runfiles. It creates a script that echos all the file names. It is useful
+for building an object that is like a cc_binary in complexity, but does not
+depend on a large toolchain.
+    """,
+    implementation = _fake_artifact_impl,
+    attrs = {
+        "deps": attr.label_list(
+            doc = "Dependencies to trigger other rules, but are then discarded.",
+            allow_files = True,
+        ),
+        "runfiles": attr.label_list(
+            doc = "Deps which are passed in DefaultInfo as runfiles.",
```

Comment:
It doesn't look like the implementation matches the documentation.  When I think of `runfiles`, I would expect to see `ctx.runfiles` be used in the rule and it be passed to the `runfiles` argument in `DefaultInfo`.  As it stands, I would just associate these as files that need to be passed to consumers.

Perhaps to make this distinction, there should be both `files` and `runfiles`, with the `files` being the what's currently in the rule?

Reply:
Good catch. I split files and runfiles to distinct attributes.

Reply:
LGTM